feature/lldp/otg_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go

```diff
@@ -0,0 +1,321 @@
+			txPackets, _ := v.Val()
+			return v.IsPresent() && txPackets != 0
+		}).Await(t)
+		if !ok {
+			txPackets, _ := lastValue.Val()
+			t.Errorf("LLDP sent packets got: %v, want: > 0.", txPackets)
+		}
+		framesIn, _ := gnmi.Watch(t, otg, gnmi.OTG().LldpInterface(lldp.Name()).Counters().FrameIn().State(), time.Minute, func(v *ygnmi.Value[uint64]) bool {
+			time.Sleep(1 * time.Second)
+			return v.IsPresent()
+		}).Await(t)
+		otgutils.LogLLDPMetrics(t, otg, c)
+		if lldpEnabled {
+			if rxPackets, _ := framesIn.Val(); rxPackets == 0 {
+				t.Errorf("LLDP received packets got: %v, want: > 0.", rxPackets)
+			}
+		} else {
+			if rxPackets, _ := framesIn.Val(); rxPackets != 0 {
+				t.Errorf("LLDP received packets got: %v, want: 0.", rxPackets)
+			}
+		}
+	}
+}
+
+// checkOtgLLDPNeighbors verifies OTG side lldp neighbor states
+func checkOtgLLDPNeighbors(t *testing.T, otg *otg.OTG, c gosnappi.Config, expLldpNeighbors map[string][]lldpNeighbors) {
+	otgutils.LogLLDPNeighborStates(t, otg, c)
+
+	for lldp, lldpNeighbors := range expLldpNeighbors {
+		lldpState := gnmi.Get(t, otg, gnmi.OTG().LldpInterface(lldp).State())
+
+		if len(lldpNeighbors) == 0 {
+			if lldpState.LldpNeighborDatabase != nil {
+				t.Errorf("LldpNeighborDatabase = %v; want nil", lldpState.LldpNeighborDatabase)
+				return
+			}
+		} else {
+			neighbors := lldpState.GetLldpNeighborDatabase().LldpNeighbor
+			for _, lldpNeighbor := range lldpNeighbors {
+				neighborFound := false
+				for _, neighbor := range neighbors {
+					if neighbor.GetChassisId() == lldpNeighbor.chassisId &&
+						neighbor.GetChassisIdType() == lldpNeighbor.chassisIdType &&
+						neighbor.GetPortId() == lldpNeighbor.portId &&
+						neighbor.GetPortIdType() == lldpNeighbor.portIdType &&
+						neighbor.GetSystemName() == lldpNeighbor.systemName {
+						neighborFound = true
+						break
+					}
+				}
+				if !neighborFound {
+					t.Errorf("LLDP Neighbor not found")
+				}
+			}
+		}
+
+	}
+}
+
+// verifyDUTTelemetry verifies the telemetry values from the node such as port LLDP neighbor info.
+func verifyDUTTelemetry(t *testing.T, dut *ondatra.DUTDevice, nodePort *ondatra.Port, conf *oc.Lldp, dutPeerState lldpNeighbors, lldpEnabled bool) {
+	verifyNodeConfig(t, dut, nodePort, conf, lldpEnabled)
+	interfacePath := gnmi.OC().Lldp().Interface(nodePort.Name())
+
+	// Ensure that DUT does not generate any LLDP messages irrespective of the
+	// configuration of lldp/interfaces/interface/config/enabled (TRUE or FALSE)
+	// on any interface.
+	var gotLen int
+	if !lldpEnabled {
+		if _, ok := gnmi.Watch(t, dut, interfacePath.State(), time.Minute, func(val *ygnmi.Value[*oc.Lldp_Interface]) bool {
+			intf, present := val.Val()
+			if !present {
+				return false
+			}
+			gotLen = len(intf.Neighbor)
+			return gotLen == 0
+		}).Await(t); !ok {
+			t.Errorf("Number of neighbors got: %d, want: 0.", gotLen)
+		}
+		return
+	}
```

Comment:
Let's split verifyDUTTelemetry into verifyDUTWithoutLLDP and verifyDUTWithLLDP.

Reply:
verifyDUTTelemetry only runs when LLDP is enabled